### PR TITLE
Upgrade Haskell server to fully support Babbage-era transactions

### PR DESCRIPTION
**Pull request by errfrom**
Closes https://github.com/Plutonomicon/cardano-transaction-lib/issues/645

1. Upgraded `POST /fees` endpoint to handle Babbage-era transactions that use the new Babbage features (for example, it's now possible to estimate fees for a transaction with collateral return)

2. Provided `Aeson` instances for `Ledger.Script` and `Ledger.Data` that were removed in this PR (https://github.com/input-output-hk/plutus/pull/4394)

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [x] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,7 +65,6 @@ This release adds support for running CTL contracts against Babbage-era nodes. *
 - Modules `Metadata.Seabug` and `Metadata.Seabug.Share`
 - `POST /eval-ex-units` Haskell server endpoint ([#665](https://github.com/Plutonomicon/cardano-transaction-lib/pull/665))
 - Truncated test fixtures for time/slots inside `AffInterface` to test time/slots not too far into the future which can be problematic during hardforks https://github.com/Plutonomicon/cardano-transaction-lib/pull/676
-- `d` and `extraEntropy` protocol parameters from protocol parameters update proposal
```
Comment (klntsky) [inline CHANGELOG.md:54]: Why is this line removed? Bad merge?

Comment (errfrom) [reply CHANGELOG.md:54]: @klntsky Probably yes, I also noticed there're some issues after merge, fixing it now

Comment (jy14898) [reply CHANGELOG.md]: @klntsky It looks like that change is in the wrong section anyway, it came after v2.0.0-alpha

**Review by ngua — DISMISSED** [review]
Sorry, this is my fault, but we need to revert `easy-purescript-nix` to `d56c436a66ec2a8a93b309c83693cef1507dca7a`. It's probably best to pass and explicit `rev` in the input. This is why the bundle is failing (a new `spago` version apparently requires network access for bundling)

**Review by ngua — APPROVED** [review]
